README.pod

```diff
@@ -6,13 +6,11 @@ App::Git::Autofixup - create fixup commits for topic branches
 
 =head1 SYNOPSIS
 
-    git-autofixup [<options>] <revision>
+    git-autofixup [<options>] [<revision>]
 
 =head1 DESCRIPTION
 
-F<git-autofixup> parses hunks of changes in the working directory out of C<git diff> output and uses C<git blame> to assign those hunks to commits in C<E<lt>revisionE<gt>..HEAD>, which will typically represent a topic branch, and then creates fixup commits to be used with C<git rebase --interactive --autosquash>. It is assumed that hunks near changes that were previously committed to the topic branch are related.
-
-C<@{upstream}> or C<@{u}> is likely a convenient value to use for C<E<lt>revisionE<gt>> if the current branch has a tracking branch. See C<git help revisions> for other ways to specify revisions.
+F<git-autofixup> parses hunks of changes in the working directory out of C<git diff> output and uses C<git blame> to assign those hunks to commits in C<E<lt>revisionE<gt>..HEAD>, which will typically represent a topic branch, and then creates fixup commits to be used with C<git rebase --interactive --autosquash>. It is assumed that hunks near changes that were previously committed to the topic branch are related. If no revision is given and the current branch has a tracking branch, then C<@{upstream}> is used to find reasonable fork-points or merge-bases to use as upstream cutoffs. See C<git help revisions> for info about how to specify revisions.
```

Comment:
> If no revision is given and the current branch has a tracking branch, then C<@{upstream}> is used to find reasonable fork-points or merge-bases to use as upstream cutoffs. See C<git help revisions> for info about how to specify revisions.

Sounds fine. It's unfortunate that Git has inconsistent terminology ("tracking branch" and "upstream"), not sure which one is better. Maybe we should use only upstream because it's in `--set-upstream-to` and `@{upstream}`.

Also we could rephrase as

    If not specified, C<E<lt>revisionE<gt>> defaults to C<git merge-base --forkpoint HEAD @{upstream} || git merge-base HEAD @{upstream}>. See C<git help revisions> for info about how to specify revisions.

which is not entirely accurate but maybe easier to understand?

Reply:
Hmm, yeah, in the `git-branch` manpage they use "upstream branch" and "tracking branch" interchangably. Good point about upstream being more prevalent in the command-line interface, but previously I thought "tracking branch" was the canonical name for this concept in git and I'm not sure I would have known exactly what someone meant by "upstream branch". So now I'm thinking it might be good to use both, like "tracking/upstream branch".

Cool, I like describing the behaviour in terms of git commands. Very clear for people familiar with the commands, and for those that aren't it's obvious what to read to figure it out.